File: java/quantile-estimator/src/main/java/org/mydotey/quantileestimator/value/LongCalculator.java
package org.mydotey.quantileestimator.value;

/**
 * @author koqizhao
 *
 * Apr 1, 2018
 */
public class LongCalculator implements Calculator<Long> {

    public static final LongCalculator DEFAULT = new LongCalculator();

    @Override
    public Long add(Long t, Long t2) {
        return t + t2;
    }

    @Override
    public Long subtract(Long t, Long t2) {
        return t - t2;
    }

    @Override
    public Long multiply(Long t, double t2) {
        return (long) (t * t2);
    }

    @Override
    public Long divide(Long t, double t2) {
        return (long) (t / t2);
    }

}

File: dotnet/src/MyDotey.QuantileEstimator/Value/LongCalculator.cs
using System;

/**
 * @author koqizhao
 *
 * Apr 1, 2018
 */
namespace MyDotey.Quantile.Value
{
    public class LongCalculator : ICalculator<long>
    {
        public static readonly LongCalculator DEFAULT = new LongCalculator();

        public virtual long Add(long t, long t2)
        {
            return t + t2;
        }

        public virtual long Subtract(long t, long t2)
        {
            return t - t2;
        }

        public virtual long Multiply(long t, double t2)
        {
            return (long)(t * t2);
        }

        public virtual long Divide(long t, double t2)
        {
            return (long)(t / t2);
        }
    }
}